### PR TITLE
fix(bpdm): add unsupported bpdm identifier #618

### PR DESCRIPTION
**Pull request by ntruchsess**
## Description

This PR contains the changes from https://github.com/eclipse-tractusx/portal-backend/pull/619 plus the required migration

## Why

see https://github.com/eclipse-tractusx/portal-backend/pull/619

## Issue

#618 

## Checklist

Please delete options that are not relevant.

- [X] I have followed the [contributing guidelines](https://github.com/eclipse-tractusx/portal-assets/blob/main/docs/developer/Technical%20Documentation/Dev%20Process/How%20to%20contribute.md#commit-and-pr-guidelines)
- [X] I have performed a self-review of my own code
- [X] I have successfully tested my changes locally
- [X] I have added tests that prove my changes work
- [X] I have checked that new and existing tests pass locally with my changes

### CODE REVIEW

File: tests/externalsystems/Bpdm.Library/BpdmServiceTests.cs [hunk 1]
```diff
@@ -238,6 +262,8 @@ public async Task FetchInputLegalEntity_WithValidResult_ReturnsExpected()
         result.Should().NotBeNull();
         result.ExternalId.Should().Be(externalId);
         result.LegalEntity?.Bpnl.Should().Be("BPNL00000007QGTF");
+        result.Identifiers.Should().ContainSingle()
```
Comment (typecastcloud) [inline tests/externalsystems/Bpdm.Library/BpdmServiceTests.cs:265]: @ntruchsess please review BPDM reference implementation for business-partner/search endpoint.

[toBusinessPartnerOutputDto](https://github.com/eclipse-tractusx/bpdm/blob/2270b8f251cb1d9d081159e590c71821ba1ad900/bpdm-gate/src/main/kotlin/org/eclipse/tractusx/bpdm/gate/service/BusinessPartnerMappings.kt#L57)

The endpoint can return a list of identifiers.
identifiers = entity.identifiers.map(::toIdentifierDto),

**Comment by jjeroch** [comment]
Thanks Norbert!
I will quickly approach the FE change description and link it here as well

**Review by Phil91 — DISMISSED** [review]

**Comment by ntruchsess** [comment]
> Thanks Norbert! I will quickly approach the FE change description and link it here as well
> 
> We should also enhance country_assigned_identifiers - to ensure that the e2e flow works we need a reference for the DUNS. Or did you already analyzed it?

@jjeroch country_assigned_identifiers must be adjusted, otherwise the new identifier is pretty useless as one cannot post it to registration-endpoint registration/application/{applicationId}/companyDetailsWithAddress
I need your input for that.

File: tests/externalsystems/Bpdm.Library/BpdmServiceTests.cs (hunk 1 above)
Comment (ntruchsess) [reply tests/externalsystems/Bpdm.Library/BpdmServiceTests.cs]: I did adjust the test-data and asserts accordingly

**Comment by ntruchsess** [comment]
obsolete, PR #619 has maintainer-write-access enabled

**Pull request closed without merging**